ledger-core/rms/any_lazy.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package rms
+
+import (
+	"encoding"
+	"fmt"
+	"reflect"
+
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+var _ GoGoSerializableWithText = &AnyLazy{}
+
+type AnyLazy struct {
+	value goGoMarshaler
+}
+
+func (p *AnyLazy) Get() LazyValue {
+	if vv, ok := p.value.(LazyValue); ok {
+		return vv
+	}
+	return LazyValue{}
+}
+
+func (p *AnyLazy) Set(v GoGoSerializable) {
+	p.value = v
```

Comment:
is it impossible to check for LazyValue when we set? Set/Get sequence returns silently empty LazyValue for incorrect type. I would have placed some runtime asserts.

Reply:
well a lazy container is usually asymmetrical - set is for not-serialized value, and read for serialized / lazy one. Will correct naming to make the difference explicit

Reply:
done
https://github.com/insolar/assured-ledger/pull/386/commits/c9461a556377796c6d9ea2d51d618c6aa19302fb